tap_mysql/sync_strategies/binlog.py

```diff
@@ -309,6 +319,8 @@ def generate_streams_map(binlog_streams):
         columns = add_automatic_properties(catalog_entry,
                                            list(catalog_entry.schema.properties.keys()))
 
+        write_schema_message(catalog_entry)
```

Comment:
This feels like a bit of a hack to me. I would think the way we'd want to do it would be to augment the [original schema production code](https://github.com/singer-io/tap-mysql/blob/1889bec05f001acd0b4ba4751a1dfee3f7573b16/tap_mysql/__init__.py#L661
), if the stream was a binlog stream, it would append these automatic fields to it. You could even do that via a keyword arg that defaulted to False and then just pass it from that binlog loop.

Reply:
i like it.